packages/provider/provider/remote.js

```diff
@@ -1,81 +1,90 @@
 
+//
+// Cache and Queue setup to prevent duplicate HTTP requests.
+//
+const queue = new Queue();
+const cache = {};
+
 /**
- * Fetch remote assets, and process them.
+ * Fetch a given resource and parse it's outcome, store it, and send it to
+ * all other requests that were waiting for the same response.
  *
- * @constructor
+ * @param {Object} options Request options.
+ * @param {AssetParser} parser AssetParser instance
+ * @param {Function} fn Error first completion callback
+ * @returns {Undefined|Boolean} Indication of cached, queued or loaded response.
  * @public
  */
-export default class Remote {
-  constructor() {
-    this.queue = new Queue();
-    this.cache = {};
+function remote(options, parser, fn) {
+  const { url, format } = options;
+  const id = queue.id('GET', url);
+  const item = cache[id];
+
+  //
+  // Fastest option first, it's already cached, or we already fetching the
+  // exact same URI.
+  //
+  if (item) {
+    debug(`the requested url(${url}) was already requested and cached`);
+    return fn(null, item);
+  }
+
+  if (queue.add('GET', url, fn)) {
+    debug(`the requested url(${url}) was already requested, waiting for callback`);
+    return true;
   }
 
   /**
-   * Fetch a given resource and parse it's outcome, store it, and send it to
-   * all other requests that were waiting for the same response.
-   *
-   * @param {Object} options Request options.
-   * @param {AssetParser} parser AssetParser instance
-   * @param {Function} fn Error first completion callback
-   * @returns {Undefined|Boolean} Indication of cached, queued or loaded response.
-   * @public
-   */
-  fetch(options, parser, fn) {
-    const { method, uri, format, timeout } = options;
-    const id = this.queue.id(method, uri);
-    const item = this.cache[id];
+  * Generic processing function for parser and loader responses.
+  *
+  * @param {Error} err Optional error.
+  * @param {Object} data Parsed data.
+  * @private
+  */
+  const loaded = (err, data) => {
+    const error = err || null;
+    const svgs = data || {};
 
     //
-    // Fastest option first, it's already cached, or we already fetching the
-    // exact same URI.
+    // Worth noting, we don't want to cache errors even if it means that
+    // multiple requests will be send. In unexpected slowness or downtime of
+    // a CDN we might be able to fetch a resource in the end. Especially
+    // important for native devices!
     //
-    if (item) {
-      debug(`the requested uri(${uri}) was already requested and cached`);
-      return fn(null, item);
-    }
+    if (!error) cache[id] = svgs;
+    queue.run('GET', url, error, svgs);
+  };
 
-    if (this.queue.add(method, uri, fn)) {
-      debug(`the requested uri(${uri}) was already requested, waiting for callback`);
-      return true;
+  debug(`first time requesting url(${url})`);
+  fetch(url)
+  .then(function status(response) {
+    if (response.status >= 200 && response.status < 300) {
```

Comment:
I was going to suggest `response.ok` but apparently [Safari](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok#Browser_compatibility) doesn't have a full response implementation. 😢